### ISSUE TITLE
Closing node does not work inside grouped empty directories

### ISSUE THREAD

**Issue by echasnovski** [kyazdani42/nvim-tree.lua#614]
Executing 'close_node' on node inside directory which is a part of "empty group" leads to jump to the top of a tree.

How to reproduce:
- Use the following Lua init file (for example, in 'init_nvim-tree.lua'):
```lua
-- Enable 'nvim-tree' functionality in suitable way
vim.cmd([[packadd nvim-tree]])

vim.g.nvim_tree_group_empty = 1
```
- Inside some directory create the following directory structure:
```
├── dir0
├── dir1
│   ├── a
│   └── dir2
│       └── dir3
│           └── c
└── init_nvim-tree.lua
```
- Run `nvim -u init_nvim-tree.lua` from inside mentioned directory.
- Inside Neovim run `:NvimTreeOpen`.
- Navidate to file 'c'.
- Hit `<BS>`. Node is not closed but cursor just jumps to the top. Expected the same outcome as if hit `<BS>` on 'dir2/dir3' node.

Here is an animation of behavior:
![nvim-tree_grouped-empty-issue](https://user-images.githubusercontent.com/24854248/132317747-0271961c-37ff-4ed7-b60e-0fbea7c188d4.gif)

**Comment by kyazdani42** [comment]
fix in `1.2.6` (84126d3eb5614b64)

**Comment by echasnovski** [comment]
@kyazdani42, thank you for the fix.
This only partially solved the problem, though. There are still problems when closing subdirectory of grouped directories.
Tree structure:
```
├── dir0
└── dir1
    ├── a
    └── dir2
        └── dir3
            ├── c
            └── dir4
                └── d
```

Animation:
![nvim-tree_issue_group-empty](https://user-images.githubusercontent.com/24854248/133133805-342dadce-024c-4f3f-bf99-f4df11547d3e.gif)

Using d7f73b5 (`1.2.8`)
